UI/backend/app.py

```diff
@@ -11,106 +15,110 @@
+        insert_time_slot(day, start_time, end_time)
     try:
-        if not course and not time:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst], capture_output=True, text=True)
-        elif not time and course:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst, course], capture_output=True, text=True)
-        elif time and course:
-            result = subprocess.run(['bash', cpp_exec_path, '--addInstructor', inst, course, time], capture_output=True, text=True)
-        else:
-            return jsonify({'success': False, 'message': 'If time is inputed course is mandatory'}), 400
-        if result.returncode == 0:
-            return jsonify({'success': True, 'message': 'Instructor added successfully!'})
-        else:
-            return jsonify({'success': False, 'message': 'Failed to add instructor', 'details': result.stderr}), 500
-    except Exception as e:
-        print(f"Exception occurred: {str(e)}")
-        return jsonify({'success': False, 'message': str(e)}), 500
+        insert_instructor(*insert_func_args)
+        insert_course(*insert_course_args)
+        return jsonify({'success': True, 'message': 'Instructor added successfully!'})
+    except Exception as db_error:
+        return jsonify({'success': False, 'message': str(db_error)}), requests.codes.internal_server_error
 
 @app.route('/schedule', methods=['GET'])
 def schedule():
-    app_dir = os.path.dirname(os.path.abspath(__file__))
-    grand_parent_dir = os.path.dirname(os.path.dirname(app_dir))
-    cpp_exec_path = os.path.join(grand_parent_dir, 'run.sh')
-
+    conn = connect_to_db()
+    cur = conn.cursor()
     try:
-        result = subprocess.run(['bash', cpp_exec_path, '--schedule'], capture_output=True, text=True)
+        cur.execute("SELECT json_build_object('courses', json_agg(row_to_json(t))) FROM (SELECT course_name, course_day, course_start_time, course_end_time FROM course) t")
+        courses_result = cur.fetchone()[0]
+        cur.execute("SELECT json_build_object('instructors', json_agg(row_to_json(t))) FROM (SELECT instructor_name, inst_course_name, inst_day, inst_start_time, inst_end_time FROM instructor) t")
+        instructors_result = cur.fetchone()[0]
+        cur.execute("SELECT json_build_object('timeSlots', json_agg(row_to_json(t))) FROM (SELECT day, start_time, end_time FROM time_slot) t")
+        time_slot_result = cur.fetchone()[0]
+    except Exception as error:
+        print("Error runing a query to create a json", error)
+    combined_result = {
+        "courses": courses_result["courses"],
+        "instructors": instructors_result["instructors"],
+        "timeSlots": time_slot_result["timeSlots"]
+    }
+
+    with open('output.json', 'w') as f:
+        json.dump(combined_result, f, indent=4)
+
+    fileToReplace = '""'
+    fileToSearch = "null"
+
+    with open('output.json', 'r') as file:
```

Comment:
I don't understand what are doing this file openings-closings here...
If this is a must, then explain

Reply:
fixed